src/dialect/postgresql.rs

```diff
@@ -20,7 +20,7 @@ impl Dialect for PostgreSqlDialect {
         // See https://www.postgresql.org/docs/11/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS
         // We don't yet support identifiers beginning with "letters with
         // diacritical marks and non-Latin letters"
-        (ch >= 'a' && ch <= 'z') || (ch >= 'A' && ch <= 'Z') || ch == '_'
+        (ch >= 'a' && ch <= 'z') || (ch >= 'A' && ch <= 'Z') || ch == '_' || ch == '$'
```

Comment:
This doesn't match the documentation linked just above this line (Postgres seems to have `$1` and `$$comments$$`, but I don't see how `$foo` is supposed to be parsed), so I'd like to split this separately. You [should be able](https://github.com/ballista-compute/sqlparser-rs/issues/48) to keep it in a dialect of your own for now.